pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -18,38 +22,82 @@ import (
+	client     configv1client.ConfigV1Interface
+	coreClient corev1client.CoreV1Interface
 
 	lock        sync.Mutex
 	lastVersion *configv1.ClusterVersion
 }
 
-func New(client configv1client.ConfigV1Interface) *Gatherer {
+func New(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface) *Gatherer {
 	return &Gatherer{
-		client: client,
+		client:     client,
+		coreClient: coreClient,
 	}
 }
 
 var reInvalidUIDCharacter = regexp.MustCompile(`[^a-z0-9\-]`)
 
 func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error {
-	return collectRecords(ctx, recorder,
-		func() ([]record.Record, []error) {
-			config, err := i.client.ClusterOperators().List(metav1.ListOptions{})
-			if err != nil {
-				return nil, []error{err}
-			}
-			records := make([]record.Record, 0, len(config.Items))
-			for i := range config.Items {
-				records = append(records, record.Record{Name: fmt.Sprintf("config/clusteroperator/%s", config.Items[i].Name), Item: ClusterOperatorAnonymizer{&config.Items[i]}})
-			}
-			return records, nil
-		},
+	return record.Collect(ctx, recorder,
+		record.Aggregate(
+			func() ([]record.Record, []error) {
+				config, err := i.client.ClusterOperators().List(metav1.ListOptions{})
+				if errors.IsNotFound(err) {
+					return nil, nil
+				}
+				if err != nil {
+					return nil, []error{err}
+				}
+				records := make([]record.Record, 0, len(config.Items))
+				for i := range config.Items {
+					records = append(records, record.Record{Name: fmt.Sprintf("config/clusteroperator/%s", config.Items[i].Name), Item: ClusterOperatorAnonymizer{&config.Items[i]}})
+				}
+
+				for _, item := range config.Items {
+					if isFailingOperator(&item) {
+						for _, namespace := range namespacesForOperator(&item) {
+							pods, err := i.coreClient.Pods(namespace).List(metav1.ListOptions{})
+							if err != nil {
+								klog.V(2).Infof("Unable to find pods in namespace %s for failing operator %s", namespace, item.Name)
+							}
+							for i := range pods.Items {
+								if isFailingPod(&pods.Items[i]) {
+									records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pods.Items[i].Namespace, pods.Items[i].Name), Item: PodAnonymizer{&pods.Items[i]}})
+								}
+							}
+						}
+					}
+				}
+
+				return records, nil
+			},
+			func() ([]record.Record, []error) {
+				nodes, err := i.coreClient.Nodes().List(metav1.ListOptions{})
+				if err != nil {
+					return nil, []error{err}
+				}
+				records := make([]record.Record, 0, len(nodes.Items))
+				for i := range nodes.Items {
+					if isFailingNode(&nodes.Items[i]) {
+						records = append(records, record.Record{Name: fmt.Sprintf("config/node/%s", nodes.Items[i].Name), Item: NodeAnonymizer{&nodes.Items[i]}})
+					}
+				}
+
+				return records, nil
+			},
+		),
 		func() (record.Record, error) {
 			config, err := i.client.ClusterVersions().Get("version", metav1.GetOptions{})
+			if errors.IsNotFound(err) {
```

Comment:
hrm, if this returns 404, it means something really bad happened to that cluster, right? don't we want to know?

Reply:
Good point, getting the errors to the reports might be a good thing to do in general. @jhjaggars suggested to keep metadata about the gathering for the parts in separate jsons, and I think that's where this errors should be recorded. Probably out of scope for this PR though.

Reply:
There is a quite a lot of duplicate code here around the `IsNotFound`, which makes me think if that shouldn't rather be in somewhere else?

Reply:
It's different per gatherer, and ultimately this error isn't important (the file won't be in the zip, which is ok).

There are a few gatherers that don't call 404, so it shouldn't be generic, and a more complex gatherer won't help understanding what it does (this code is intended to be simple and easy to reason about).